### PR TITLE
Review FT

### PR DESCRIPTION
**Pull request by julian-upc**

### CODE REVIEW

**Comment by julian-upc** [comment]
So am I correct in reading that you are processing one-dimensional Gale diagrams, i.e., 4-polytopes on 7 vertices?

**Comment by TimWeenink** [comment]
Initially we thought that we had to work with 4-polytopes on 7 vertices, which is why we tried to construct the Gale diagram using that amount of points. Later Simon told us that we had to work with 4-polytopes on 8 vertices, which is why we switched from 7 to 8 in our new function checkNB.

**Comment by julian-upc** [comment]
ok. How much time do you estimate that you would need to (a) complete your code for n=7, (b) for n=8?

**Comment by fvandijken** [comment]
Julian, what are your thoughts on this assignment. Would you still give us chance to pass if we hand in a complete assignment? To finish it we would need one more week for sure due to other exams. But we are having a hard time figuring out how to solve the assignment in general.

**Comment by julian-upc** [comment]
Sorry, we have to close grades this week. I really appreciate your efforts, and I hope that the things you learned will help you in the future.

You can look at the code in the other projects, in the `review*` branches,

**Comment by fvandijken** [comment]
Julian, could you please give us the opportunity to work on the code until the deadline and then review it one more time to see if we can get a higher mark than we would have now?

**Comment by julian-upc** [comment]
ok, sure! But I have to confer with Vera about your grades, so I'll have to give you Friday 12am as a hard deadline.

Best,
Julian

**Comment by TimWeenink** [comment]
Thank you for giving us this opportunity! Btw, just to be sure, this is 12 o'clock midnight, right?

**Comment by julian-upc** [comment]
No, sorry, 12 am ... Vera and I have to agree on the grades, and enter them by Friday :/

I'm giving you the absolutely greatest margin available, up to the point that you can look at the solutions by the other team; it's actually verging on being unfair to the others... I hope you learn something!

File: 2016/exercises/final_project/Franka-Tim/poging2.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+
+goodCombs = checkCombinations()
+len(goodCombs)
+
+#Generate the coordinates at random and append a correct signlist (index as input) to generate a Gale Diagram
+
+def generateRandomGale(integer):
+    randomGD = list()
+    goodCombs = checkCombinations()
```
Comment (julian-upc) [inline 2016/exercises/final_project/Franka-Tim/poging2.py]: you do this **every time** you want to generate a random diagram!

File: 2016/exercises/final_project/Franka-Tim/poging2.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    randomGD.append([random.randrange(-5, 5) for _ in range(0, 8)])
+    randomGD.append([random.randrange(-5, 5) for _ in range(0, 8)])
+    k = goodCombs[integer]
```
Comment (julian-upc) [inline 2016/exercises/final_project/Franka-Tim/poging2.py]: you should maybe do some range checking on `integer`; you don't actually know beforehand how many elements `goodCombs` has

File: 2016/exercises/final_project/Franka-Tim/poging2.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+#Transpose matrix
+uitkomst = generateRandomGale(4) #Example to check if it works
+randomGDt = map(list,map(None,*uitkomst))
+
+#generate a random coordinate and a random correct signlist to generate a random gale diagram
+
+def mainFunction():
+    neighbourlyGD = list()
+    for i in [0..5]: #Generate 500 random points
+        rand = random.randint(0,125) #Pick a random sign configuration
```
Comment (julian-upc) [inline 2016/exercises/final_project/Franka-Tim/poging2.py]: I don't understand the magic number `125`

**Comment by julian-upc** [comment]
I imported the file into sage and ran `print mainFunction()` on it. This gave me a large list of the form
```
[[-3, -5, '-'], [-1, 1, '-'], [-2, -3, '+'], [2, -4, '-'], [0, 4, '+'],
[0, -2, '+'], [0, 4, '-'], [2, 4, '-']]
([-3, -5, '-'], [-1, 1, '-'], [-2, -3, '+'])
([-3, -5, '-'], [-1, 1, '-'], [2, -4, '-'])
([-3, -5, '-'], [-1, 1, '-'], [0, 4, '+'])
([-3, -5, '-'], [-1, 1, '-'], [0, -2, '+'])
...
([0, 4, '+'], [0, 4, '-'], [2, 4, '-'])
([0, -2, '+'], [0, 4, '-'], [2, 4, '-'])
```
and the answer `[]`. How do I interpret this?

File: 2016/exercises/final_project/Franka-Tim/poging2.py (hunk 3 above)
Comment (TimWeenink) [inline 2016/exercises/final_project/Franka-Tim/poging2.py]: We've put all indices corresponding to 'good' (i.e. #+ in the configuration = 3 or 4) sign configurations in a list, which had length 126

**Comment by TimWeenink** [comment]
The 8x3-matrices are the random Gale diagrams that we generate (print(randomGDt)). The other output is something that I used to try and find the error in the function checkSimplicial, but forgot to put as a comment.

**Comment by julian-upc** [comment]
But they don't seem to be 8x3 matrices, except for the first one which is 3x8. This is the whole output:
```
print mainFunction()
[[-3, -5, '-'], [-1, 1, '-'], [-2, -3, '+'], [2, -4, '-'], [0, 4, '+'],
[0, -2, '+'], [0, 4, '-'], [2, 4, '-']]
([-3, -5, '-'], [-1, 1, '-'], [-2, -3, '+'])
([-3, -5, '-'], [-1, 1, '-'], [2, -4, '-'])
([-3, -5, '-'], [-1, 1, '-'], [0, 4, '+'])
([-3, -5, '-'], [-1, 1, '-'], [0, -2, '+'])
([-3, -5, '-'], [-1, 1, '-'], [0, 4, '-'])
([-3, -5, '-'], [-1, 1, '-'], [2, 4, '-'])
([-3, -5, '-'], [-2, -3, '+'], [2, -4, '-'])
([-3, -5, '-'], [-2, -3, '+'], [0, 4, '+'])
([-3, -5, '-'], [-2, -3, '+'], [0, -2, '+'])
([-3, -5, '-'], [-2, -3, '+'], [0, 4, '-'])
([-3, -5, '-'], [-2, -3, '+'], [2, 4, '-'])
([-3, -5, '-'], [2, -4, '-'], [0, 4, '+'])
([-3, -5, '-'], [2, -4, '-'], [0, -2, '+'])
([-3, -5, '-'], [2, -4, '-'], [0, 4, '-'])
([-3, -5, '-'], [2, -4, '-'], [2, 4, '-'])
([-3, -5, '-'], [0, 4, '+'], [0, -2, '+'])
([-3, -5, '-'], [0, 4, '+'], [0, 4, '-'])
([-3, -5, '-'], [0, 4, '+'], [2, 4, '-'])
([-3, -5, '-'], [0, -2, '+'], [0, 4, '-'])
([-3, -5, '-'], [0, -2, '+'], [2, 4, '-'])
([-3, -5, '-'], [0, 4, '-'], [2, 4, '-'])
([-1, 1, '-'], [-2, -3, '+'], [2, -4, '-'])
([-1, 1, '-'], [-2, -3, '+'], [0, 4, '+'])
([-1, 1, '-'], [-2, -3, '+'], [0, -2, '+'])
([-1, 1, '-'], [-2, -3, '+'], [0, 4, '-'])
([-1, 1, '-'], [-2, -3, '+'], [2, 4, '-'])
([-1, 1, '-'], [2, -4, '-'], [0, 4, '+'])
([-1, 1, '-'], [2, -4, '-'], [0, -2, '+'])
([-1, 1, '-'], [2, -4, '-'], [0, 4, '-'])
([-1, 1, '-'], [2, -4, '-'], [2, 4, '-'])
([-1, 1, '-'], [0, 4, '+'], [0, -2, '+'])
([-1, 1, '-'], [0, 4, '+'], [0, 4, '-'])
([-1, 1, '-'], [0, 4, '+'], [2, 4, '-'])
([-1, 1, '-'], [0, -2, '+'], [0, 4, '-'])
([-1, 1, '-'], [0, -2, '+'], [2, 4, '-'])
([-1, 1, '-'], [0, 4, '-'], [2, 4, '-'])
([-2, -3, '+'], [2, -4, '-'], [0, 4, '+'])
([-2, -3, '+'], [2, -4, '-'], [0, -2, '+'])
([-2, -3, '+'], [2, -4, '-'], [0, 4, '-'])
([-2, -3, '+'], [2, -4, '-'], [2, 4, '-'])
([-2, -3, '+'], [0, 4, '+'], [0, -2, '+'])
([-2, -3, '+'], [0, 4, '+'], [0, 4, '-'])
([-2, -3, '+'], [0, 4, '+'], [2, 4, '-'])
([-2, -3, '+'], [0, -2, '+'], [0, 4, '-'])
([-2, -3, '+'], [0, -2, '+'], [2, 4, '-'])
([-2, -3, '+'], [0, 4, '-'], [2, 4, '-'])
([2, -4, '-'], [0, 4, '+'], [0, -2, '+'])
([2, -4, '-'], [0, 4, '+'], [0, 4, '-'])
([2, -4, '-'], [0, 4, '+'], [2, 4, '-'])
([2, -4, '-'], [0, -2, '+'], [0, 4, '-'])
([2, -4, '-'], [0, -2, '+'], [2, 4, '-'])
([2, -4, '-'], [0, 4, '-'], [2, 4, '-'])
([0, 4, '+'], [0, -2, '+'], [0, 4, '-'])
([0, 4, '+'], [0, -2, '+'], [2, 4, '-'])
([0, 4, '+'], [0, 4, '-'], [2, 4, '-'])
([0, -2, '+'], [0, 4, '-'], [2, 4, '-'])
[]
```

**Comment by TimWeenink** [comment]
When I run mainFunction(), I get the following output:
[[0, -4, '+'], [-5, -2, '-'], [-4, 3, '+'], [4, -4, '+'], [0, -1, '-'], [2, 2, '-'], [-5, -1, '-'], [-3, -3, '-']]
([0, -4, '+'], [-5, -2, '-'], [-4, 3, '+'])
([0, -4, '+'], [-5, -2, '-'], [4, -4, '+'])
([0, -4, '+'], [-5, -2, '-'], [0, -1, '-'])
([0, -4, '+'], [-5, -2, '-'], [2, 2, '-'])
([0, -4, '+'], [-5, -2, '-'], [-5, -1, '-'])
([0, -4, '+'], [-5, -2, '-'], [-3, -3, '-'])
([0, -4, '+'], [-4, 3, '+'], [4, -4, '+'])
([0, -4, '+'], [-4, 3, '+'], [0, -1, '-'])
([0, -4, '+'], [-4, 3, '+'], [2, 2, '-'])
([0, -4, '+'], [-4, 3, '+'], [-5, -1, '-'])
([0, -4, '+'], [-4, 3, '+'], [-3, -3, '-'])
([0, -4, '+'], [4, -4, '+'], [0, -1, '-'])
([0, -4, '+'], [4, -4, '+'], [2, 2, '-'])
([0, -4, '+'], [4, -4, '+'], [-5, -1, '-'])
([0, -4, '+'], [4, -4, '+'], [-3, -3, '-'])
([0, -4, '+'], [0, -1, '-'], [2, 2, '-'])
([0, -4, '+'], [0, -1, '-'], [-5, -1, '-'])
([0, -4, '+'], [0, -1, '-'], [-3, -3, '-'])
([0, -4, '+'], [2, 2, '-'], [-5, -1, '-'])
([0, -4, '+'], [2, 2, '-'], [-3, -3, '-'])
([0, -4, '+'], [-5, -1, '-'], [-3, -3, '-'])
([-5, -2, '-'], [-4, 3, '+'], [4, -4, '+'])
([-5, -2, '-'], [-4, 3, '+'], [0, -1, '-'])
([-5, -2, '-'], [-4, 3, '+'], [2, 2, '-'])
([-5, -2, '-'], [-4, 3, '+'], [-5, -1, '-'])
([-5, -2, '-'], [-4, 3, '+'], [-3, -3, '-'])
([-5, -2, '-'], [4, -4, '+'], [0, -1, '-'])
([-5, -2, '-'], [4, -4, '+'], [2, 2, '-'])
([-5, -2, '-'], [4, -4, '+'], [-5, -1, '-'])
([-5, -2, '-'], [4, -4, '+'], [-3, -3, '-'])
([-5, -2, '-'], [0, -1, '-'], [2, 2, '-'])
([-5, -2, '-'], [0, -1, '-'], [-5, -1, '-'])
([-5, -2, '-'], [0, -1, '-'], [-3, -3, '-'])
([-5, -2, '-'], [2, 2, '-'], [-5, -1, '-'])
([-5, -2, '-'], [2, 2, '-'], [-3, -3, '-'])
([-5, -2, '-'], [-5, -1, '-'], [-3, -3, '-'])
([-4, 3, '+'], [4, -4, '+'], [0, -1, '-'])
([-4, 3, '+'], [4, -4, '+'], [2, 2, '-'])
([-4, 3, '+'], [4, -4, '+'], [-5, -1, '-'])
([-4, 3, '+'], [4, -4, '+'], [-3, -3, '-'])
([-4, 3, '+'], [0, -1, '-'], [2, 2, '-'])
([-4, 3, '+'], [0, -1, '-'], [-5, -1, '-'])
([-4, 3, '+'], [0, -1, '-'], [-3, -3, '-'])
([-4, 3, '+'], [2, 2, '-'], [-5, -1, '-'])
([-4, 3, '+'], [2, 2, '-'], [-3, -3, '-'])
([-4, 3, '+'], [-5, -1, '-'], [-3, -3, '-'])
([4, -4, '+'], [0, -1, '-'], [2, 2, '-'])
([4, -4, '+'], [0, -1, '-'], [-5, -1, '-'])
([4, -4, '+'], [0, -1, '-'], [-3, -3, '-'])
([4, -4, '+'], [2, 2, '-'], [-5, -1, '-'])
([4, -4, '+'], [2, 2, '-'], [-3, -3, '-'])
([4, -4, '+'], [-5, -1, '-'], [-3, -3, '-'])
([0, -1, '-'], [2, 2, '-'], [-5, -1, '-'])
([0, -1, '-'], [2, 2, '-'], [-3, -3, '-'])
([0, -1, '-'], [-5, -1, '-'], [-3, -3, '-'])
([2, 2, '-'], [-5, -1, '-'], [-3, -3, '-'])
[[1, 0, '+'], [0, 0, '-'], [3, -4, '-'], [-1, -2, '+'], [4, 2, '-'], [-4, -4, '+'], [4, -4, '+'], [-4, 2, '-']]
([1, 0, '+'], [0, 0, '-'], [3, -4, '-'])
([1, 0, '+'], [0, 0, '-'], [-1, -2, '+'])
([1, 0, '+'], [0, 0, '-'], [4, 2, '-'])
([1, 0, '+'], [0, 0, '-'], [-4, -4, '+'])
([1, 0, '+'], [0, 0, '-'], [4, -4, '+'])
([1, 0, '+'], [0, 0, '-'], [-4, 2, '-'])
([1, 0, '+'], [3, -4, '-'], [-1, -2, '+'])
([1, 0, '+'], [3, -4, '-'], [4, 2, '-'])
([1, 0, '+'], [3, -4, '-'], [-4, -4, '+'])
([1, 0, '+'], [3, -4, '-'], [4, -4, '+'])
([1, 0, '+'], [3, -4, '-'], [-4, 2, '-'])
([1, 0, '+'], [-1, -2, '+'], [4, 2, '-'])
([1, 0, '+'], [-1, -2, '+'], [-4, -4, '+'])
([1, 0, '+'], [-1, -2, '+'], [4, -4, '+'])
([1, 0, '+'], [-1, -2, '+'], [-4, 2, '-'])
([1, 0, '+'], [4, 2, '-'], [-4, -4, '+'])
([1, 0, '+'], [4, 2, '-'], [4, -4, '+'])
([1, 0, '+'], [4, 2, '-'], [-4, 2, '-'])
([1, 0, '+'], [-4, -4, '+'], [4, -4, '+'])
([1, 0, '+'], [-4, -4, '+'], [-4, 2, '-'])
([1, 0, '+'], [4, -4, '+'], [-4, 2, '-'])
([0, 0, '-'], [3, -4, '-'], [-1, -2, '+'])
([0, 0, '-'], [3, -4, '-'], [4, 2, '-'])
([0, 0, '-'], [3, -4, '-'], [-4, -4, '+'])
([0, 0, '-'], [3, -4, '-'], [4, -4, '+'])
([0, 0, '-'], [3, -4, '-'], [-4, 2, '-'])
([0, 0, '-'], [-1, -2, '+'], [4, 2, '-'])
([0, 0, '-'], [-1, -2, '+'], [-4, -4, '+'])
([0, 0, '-'], [-1, -2, '+'], [4, -4, '+'])
([0, 0, '-'], [-1, -2, '+'], [-4, 2, '-'])
([0, 0, '-'], [4, 2, '-'], [-4, -4, '+'])
([0, 0, '-'], [4, 2, '-'], [4, -4, '+'])
([0, 0, '-'], [4, 2, '-'], [-4, 2, '-'])
([0, 0, '-'], [-4, -4, '+'], [4, -4, '+'])
([0, 0, '-'], [-4, -4, '+'], [-4, 2, '-'])
([0, 0, '-'], [4, -4, '+'], [-4, 2, '-'])
([3, -4, '-'], [-1, -2, '+'], [4, 2, '-'])
([3, -4, '-'], [-1, -2, '+'], [-4, -4, '+'])
([3, -4, '-'], [-1, -2, '+'], [4, -4, '+'])
([3, -4, '-'], [-1, -2, '+'], [-4, 2, '-'])
([3, -4, '-'], [4, 2, '-'], [-4, -4, '+'])
([3, -4, '-'], [4, 2, '-'], [4, -4, '+'])
([3, -4, '-'], [4, 2, '-'], [-4, 2, '-'])
([3, -4, '-'], [-4, -4, '+'], [4, -4, '+'])
([3, -4, '-'], [-4, -4, '+'], [-4, 2, '-'])
([3, -4, '-'], [4, -4, '+'], [-4, 2, '-'])
([-1, -2, '+'], [4, 2, '-'], [-4, -4, '+'])
([-1, -2, '+'], [4, 2, '-'], [4, -4, '+'])
([-1, -2, '+'], [4, 2, '-'], [-4, 2, '-'])
([-1, -2, '+'], [-4, -4, '+'], [4, -4, '+'])
([-1, -2, '+'], [-4, -4, '+'], [-4, 2, '-'])
([-1, -2, '+'], [4, -4, '+'], [-4, 2, '-'])
([4, 2, '-'], [-4, -4, '+'], [4, -4, '+'])
([4, 2, '-'], [-4, -4, '+'], [-4, 2, '-'])
([4, 2, '-'], [4, -4, '+'], [-4, 2, '-'])
([-4, -4, '+'], [4, -4, '+'], [-4, 2, '-'])
[[[0, -4, '+'], [-5, -2, '-'], [-4, 3, '+'], [4, -4, '+'], [0, -1, '-'], [2, 2, '-'], [-5, -1, '-'], [-3, -3, '-']]]
This is still strange, because I'd expect that it would terminate after printing 6 Gale diagrams (and corresponding combinations of length 3)

**Pull request merged**